src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const stateHistory = [];
+
+  for (const action in actions) {
+    const actionToDo = actions[action];
+
+    switch (actionToDo.type) {
+      case 'addProperties':
+        Object.assign(newState, actionToDo.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key in actionToDo.keysToRemove) {
+          const toDelete = actionToDo.keysToRemove[key];
+
+          delete newState[toDelete];
+        }
```

Comment:
Same here. See how easier it becomes with `for .. of` loop.
```suggestion
        for (const key of actionToDo.keysToRemove) {
          delete newState[key];
        }
```